server/src/main/java/org/opensearch/index/shard/IndexingStats.java

```diff
@@ -188,6 +208,11 @@ public Stats(StreamInput in) throws IOException {
             }
         }
 
+        /**
+         * This constructor will be deprecated in 4.0
```

Comment:
nit: This PR will deprecate this constructor so this is not entirely accurate. Maybe update it to `This constructor will be deprecated starting in version 3.3.0, use IndexingStats.Stats.Builder instead.`?